b2sdk/encryption/setting.py

```diff
@@ -66,17 +113,39 @@ def add_to_upload_headers(self, headers):
         elif self.mode == EncryptionMode.SSE_B2:
             headers['X-Bz-Server-Side-Encryption'] = self.algorithm.name
         elif self.mode == EncryptionMode.SSE_C:
+            if self.key.secret is None:
+                raise ValueError('Cannot use an unknown key in upload headers')
             headers['X-Bz-Server-Side-Encryption-Customer-Algorithm'] = self.algorithm.name
-            headers['X-Bz-Server-Side-Encryption-Customer-Key'] = self.key
-            headers['X-Bz-Server-Side-Encryption-Customer-Key-Md5'] = hex_md5_of_bytes(self.key)
+            headers['X-Bz-Server-Side-Encryption-Customer-Key'] = self.key.key_b64()
+            headers['X-Bz-Server-Side-Encryption-Customer-Key-Md5'] = self.key.key_md5()
+            headers['X-Bz-Info-sse_c_key_id'] = self.key.id
```

Comment:
okay, but `key.key_id` actually comes from the user directly (except for sync operations, where it comes from a provider, which comes from a user). So the user has absolute control over what key_id will be inserted. To solve the file-info-size-limit problem, maybe `null` keys should just not be sent (not added to headers) and that's it?

Reply:
`null` keys should not be sent, but that doesn't solve the problem. Someone can put sse_c_key_id in file info and then we try to put a similar thing in - they should get a meaningful error message instead of undefined behavior the server might take (accept first header of this name, or second, or both, or random?)